rust/template/observe/observer.rs

```diff
@@ -1,8 +1,9 @@
+use std::fmt::Debug;
 use std::sync::Arc;
 use std::sync::Mutex;
 
 /// A trait for objects that can observe an observable one.
-pub trait Observer<T, E>: Send
+pub trait Observer<T, E>: Debug + Send
```

Comment:
We should rename this class to perhaps TransactionObserver.
The Observer interface must have an on_next method.

Reply:
Can you explain why we need an `on_next` method?

Reply:
I didn't say we need an on_next method, I said that the Observer pattern must have an on_next method. See e.g., https://www.journaldev.com/22594/rxjava-observables-observers.
So this is not a standard Observer, so we should call it something else.

Reply:
I see. Thanks for the link. That makes it a bit more clear. I tend to disagree, though. Just because some Java library implements an Observer which has an `on_next` method does neither mean that the name `Observer` is reserved nor that everybody has an association/expectation that just because it's called `Observer` it implements exactly this interface. My understanding is that observer is a [generic design pattern](https://en.wikipedia.org/wiki/Observer_pattern) with a wide variety of implementations and I don't see us claiming any relationship to or compatibility with `RxJava`.
So honestly, I'd like to keep the name this concise until we have multiple observers that require differentiation. Or am I missing something?

Reply:
It is true that `TransactionObserver` may be a more accurate name for this interface, not because of `on_next()` method not being part of it, but because the interface groups updates in batches punctuated by `on_start`, `on_commit`, or whatever those methods are called. But `TransactionObserver` is a bit verbose and so I prefer sticking to `Observer`.